mlflow/cli.py

```diff
@@ -566,12 +587,51 @@ def gc(older_than, backend_store_uri, run_ids):
     else:
         run_ids = run_ids.split(",")
 
-    for run_id in run_ids:
+    if not skip_experiments:
+        deleted_older_experiment_ids = []
+        next_experiment_page_token = None
+        time_threshold = get_current_time_millis() - time_delta
+        filter_string = f"last_update_time < {time_threshold}" if older_than else ""
+        while True:
+            page_results = backend_store.search_experiments(
+                view_type=ViewType.DELETED_ONLY,
+                filter_string=filter_string,
+                page_token=next_experiment_page_token,
+            )
+            for experiment in page_results:
+                deleted_older_experiment_ids.append(experiment.experiment_id)
+            if page_results.token is None:
+                break
+            next_experiment_page_token = page_results.token
+
+    if not experiment_ids:
+        experiment_ids = deleted_older_experiment_ids
```

Comment:
`deleted_older_experiment_ids` is only defined when `skip_experiments` is False.